src/nvim/ops.c

```diff
@@ -2797,7 +2797,7 @@ static void op_yank_reg(oparg_T *oap, bool message, yankreg_T *reg, bool append)
     xfree(reg->y_array);
   }
 
-  if (message) {  // Display message about yank?
+  if (message && p_ch > 0) {  // Display message about yank?
```

Comment:
I suppose we can do this for now, but we'll want to revert this later when we have a "never show press-enter" feature. So the message would still display but the hit-enter prompt is avoided.

Reply:
OK.